### PR TITLE
copyblocks: add option to bypass duration check for no-compact blocks

### PR DESCRIPTION
**Pull request by andyasp**
#### What this PR does

Adds a flag to bypass the block duration check in `copyblocks` for blocks marked as no-compact.

It's a common pattern to use `min-block-duration` with the intent of targeting blocks that are fully compacted. This can miss blocks that are marked as `no-compact` that have a shorter duration. This flag makes it easier to not miss those blocks.

#### Which issue(s) this PR fixes or relates to

Fixes N/A

#### Checklist

- [ ] Tests updated.
- [x] Documentation added.
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [ ] [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

**Comment by narqo** [comment]
The CHANGELOG has just been cut to prepare for the next release. Please rebase `main` and eventually move the CHANGELOG entry added / updated in this PR to the top of the CHANGELOG.md document. Thanks!

**Comment by andyasp** [comment]
I struggled with the name as well. Open to ideas for it and I'll hold off while trying to think of something myself.

File: tools/copyblocks/main.go
Comment (narqo) [inline tools/copyblocks/main.go:63]: How about `--skip-no-compact-block-duration-check`?

Comment (andyasp) [reply tools/copyblocks/main.go:63]: That is way more clear. I'll change it to that, thanks!

File: tools/copyblocks/README.md
Comment (andyasp) [inline tools/copyblocks/README.md:14]:
```suggestion
- Configurable minimum block duration (`--min-block-duration`) and (`--skip-no-compact-block-duration-check`) to target blocks that are not awaiting compaction
```

**Review by narqo — APPROVED** [review]
🔥 Works for me

File: CHANGELOG.md
Comment (andyasp) [inline CHANGELOG.md:50]:
```suggestion
* [ENHANCEMENT] `copyblocks`: Added `--skip-no-compact-block-duration-check`, which defaults to `false`, to simplify targeting blocks that are not awaiting compaction. #9439
```

Comment (andyasp) [reply CHANGELOG.md]: I'll merge after fixing this last shameful miss.

**Pull request merged**
